client/src/components/layout/ReviewTile.js

```diff
@@ -1,7 +1,21 @@
-import React, { useState } from "react"
+import React, { useEffect, useReducer, useState } from "react"
 import ReviewEditButton from "./ReviewEditButton"
 import ReviewDeleteButton from "./ReviewDeleteButton"
 const ReviewTile = ({ review, user, setPlanet, planet }) => {
+  const [currentUserVoteValue, setCurrentUserVoteValue] = useState(0)
+  const retrieveCurrentUserVoteState = () => {
+    const initialCurrentUserVoteValue = 0
+    review.votes.forEach((vote) => {
+      if (user.id === vote.userId) {
+        setCurrentUserVoteValue(vote.voteValue)
+      }
+    })
+  }
```

Comment:
there is no state change in this, so it doesnt need to go into useEffect. If you would like, you can make it a service function like `translateServerErrors` and then import it and call it as needed. But no need for useEffect

Reply:
Im also realizing this logic should go into your serializer, such that every review comes with the voteValue for the current user!